Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -359,20 +354,25 @@ int RunVertices(int vtx_attr_group, OpcodeDecoder::Primitive primitive, int coun
     // Doing early return for the opposite case would be cleaner
     // but triggers a false unreachable code warning in MSVC debug builds.
 
-    CheckCPConfiguration(vtx_attr_group);
+    if (g_needs_cp_xf_consistency_check) [[unlikely]]
+    {
+        g_needs_cp_xf_consistency_check = false;
+        CheckCPConfiguration(vtx_attr_group);
+    }
```

Comment:
I'm surprised that this makes a difference; I'd expect `CheckCPConfiguration` to be inlined and thus the same code to be generated in both cases.

Reply:
To clarify, though, I haven't actually *checked* if it behaves that way; if you have data one way or another let me know. I don't have a strong opinion between the two forms.

Reply:
MSVC doesn't seem to inline CheckCPConfiguration.

Reply:
Alright, in that case it's probably better to use `[[unlikely]]` and check in `RunVertices`; based on my understanding `[[likely]]` in a non-inlined `CheckCPConfiguration` would still pay the cost of a function call and then return while `[[unlikely]]` in `RunVertices` would skip that. Forcing inlining wouldn't really help much in either case. So the way you originally changed it should be good (apart from the incorrect indentation).

Reply:
> based on my understanding [[likely]] in a non-inlined CheckCPConfiguration would still pay the cost of a function call and then return while [[unlikely]] in RunVertices would skip that

Yeah, that's the idea.